File: logic/items/goldenApple.js
const Item = require("./item");

class GoldenApple extends Item {
  static config = {
    type: "golden-apple",
    affect: "self",
    pickUpReward: 70,
    duration: 5,
    spawnWeight: 7,
    symbol: "G",
  };

  /**
   * Creates a new item instance
   * @param {Object} position - The position of the item
   * @param {number} position.row - The row coordinate of the item
   * @param {number} position.col - The column coordinate of the item
   */
  constructor(position) {
    super(position, GoldenApple.config);
  }

  /**
   * Implements the effect of picking up a golden apple
   * @param {Player} player - The player that collided with the item
   */
  do(player) {
    // add a segment to player tail to be immediately removed by pop()
    player.body.push(player.body[player.body.length - 1]);
  }
}

module.exports = GoldenApple;

File: simple_snake/simple_snake_logic/items/goldenApple.py
# items/golden_apple.py
from .item import Item

class GoldenApple(Item):
    config = {
        'type': 'golden-apple',
        'affect': 'self',
        'pickUpReward': 70,
        'duration': 5,
        'spawnWeight': 7,
        'symbol': 'G'
    }

    def __init__(self, position, affect):
        super().__init__(position, GoldenApple.config)

    def do(self, player):
        """
        Implementira efekt skupljanja zlatne jabuke.
        """
        if player.body:
            player.body.append(player.body[-1])